src/mscorlib/shared/System/ReadOnlyMemory.cs

```diff
@@ -125,11 +141,21 @@ internal ReadOnlyMemory(OwnedMemory<T> owner, int index, int length)
         public ReadOnlyMemory<T> Slice(int start)
         {
             if ((uint)start > (uint)_length)
-                ThrowHelper.ThrowArgumentOutOfRangeException();
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.start);
+            }
 
             if (_index < 0)
```

Comment:
I think you can just do `return new ReadOnlyMemory<T>(_object, _index + start, _length - start)` here for all cases.

Reply:
Good point.  Will simplify.